### PR TITLE
sort schools by shortest school name first

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
sort by shortest length first. This works well for finding the closest matched text

### CODE REVIEW

**Comment by dsnyder-fairway** [comment]
downside here is that searching for part of a school name may lead to still not finding the school you were looking for. On the other hand, this doesn't introduce that issue

File: webapp/src/main/webapp/src/app/school-grade/results/school-results.component.ts [hunk 1]
```diff
@@ -155,10 +157,10 @@ export class SchoolResultsComponent implements OnInit {
...
                 this.organizationService.searchSchoolsWithDistrictsBySchoolName(search)
                   .pipe(map(
-                    (organizations: any[]) =>
-                      organizations.filter(
-                        organization => this.organizations.findIndex(x => organization.equals(x)) === -1
-                      ))
+                    (schools: School[]) =>
+                      schools.filter(
+                        (school: School) => this.organizations.findIndex(x => school.equals(x)) === -1
+                      ).sort(ordering(byNumber).on<School>(s => s.name.length).compare))
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/school-grade/results/school-results.component.ts]: i think ideally we want to do this on the backend no?
could be wrong but i think there is the case where the exact match is not within the natural 20 first results so it doesnt even make its way to the frontend right @wtritch?

File: webapp/src/main/webapp/src/app/shared/organization/organization.service.ts [hunk 2]
```diff
@@ -27,17 +27,15 @@ export class OrganizationService {
...
-          limit: limit
-        }
-      });
-    } else {
-      return this.dataService.get(`${ReportingServiceRoute}/organizations/schools`, {
-        params: {}
-      });
+      options.params = {
+        limit: limit
+      };
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/organization/organization.service.ts]: i think you can just always pass `{params: {limit: limit}}` and spring will consider `limit: undefined` to be an empty/missing param

File: webapp/src/main/webapp/src/app/school-grade/results/school-results.component.ts (hunk 1 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/school-grade/results/school-results.component.ts]: That seems correct Landon. This should happen on the backend.

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/school-grade/results/school-results.component.ts]: done (moved to the query)

File: webapp/src/main/webapp/src/app/shared/organization/organization.service.ts (hunk 2 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/organization/organization.service.ts]: you were correct. this has been updated

File: webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts [hunk 3]
```diff
@@ -68,7 +68,7 @@ export class SBTypeahead implements OnInit {
           .sort(
             join(
               ordering(byString).on<Option>(o => o.group === undefined ? '' : o.group).compare,
-              ordering(byString).on<Option>(o => o.label).compare
+              ordering(byNumber).on<Option>(o => o.label.length).compare
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts]: If the back-end is sorting the results, I wouldn't sort on the front-end as well.

File: common/src/main/resources/common.sql.yml [hunk 4]
```diff
@@ -66,6 +66,7 @@ sql:
         WHERE 1=:allow_school
           AND (:name_query IS NULL OR school.name LIKE concat('%', :name_query, '%'))
           AND (1=:statewide OR school.district_group_id IN (:district_group_ids) OR school.district_id IN (:district_ids) OR school.school_group_id IN (:school_group_ids) OR school.id IN (:school_ids))
+        ORDER BY LENGTH(name)
```
Comment (wtritch) [inline common/src/main/resources/common.sql.yml]: Maybe add a validating integration test to verify that aaa comes before aaaa comes before etc?

File: webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts]: ah you are right. Forgot this hit the same endpoint :-/

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts]: hmmm. Removing this line, or the entire sort makes the ordering not pretty.

File: common/src/main/resources/common.sql.yml (hunk 4 above)
Comment (dsnyder-fairway) [reply common/src/main/resources/common.sql.yml]: sounds good

File: webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts]: Oh I see...the Controller returns a set

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/form/sb-typeahead.component.ts]: done

File: common/src/test/java/org/opentestsystem/rdw/reporting/common/repository/impl/ReportingJdbcOrganizationRepositoryIT.java [hunk 5]
```diff
@@ -101,9 +101,10 @@ public void itShouldFilterDistrictsByPermissions() {
     public void itShouldFilterSchoolGroupsByPermissions() {
         final Collection<Organization> allOrgs = repository.findAll(schoolGroups(-10L), allOrgsQuery);
         assertThat(allOrgs.stream().map(Organization::getName))
-                .containsOnly(
-                        "schoolgroup1",
-                        "school1");
+                .containsExactly(
+                        "school1",
+                        "schoolgroup1"
+                        );
```
Comment (dsnyder-fairway) [inline common/src/test/java/org/opentestsystem/rdw/reporting/common/repository/impl/ReportingJdbcOrganizationRepositoryIT.java]: @wtritch some changes here to reflect the ordering. I went with these two tests as these have distinct name lengths. When the name length is the same, 2 may precede 1.

**Pull request merged**